firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -3721,6 +3728,18 @@ extension BrowserViewController: TabManagerDelegate {
                 // The webView can go gray if it was zombified due to memory pressure.
                 // When this happens, the URL is nil, so try restoring the page upon selection.
                 selectedTab.reload()
+                // If the webView is loading, we hide the lock
+                // icon and wait for did finish to get the lasted secure content status
+            } else if webView.isLoading {
+                if !isToolbarRefactorEnabled {
+                    self.urlBar.locationView.hideTrackingProtectionButton()
+                }
+                // If not, we show the lock icon with the secure content status of the webView
+            } else {
+                if !isToolbarRefactorEnabled {
+                    self.urlBar.locationView.hasSecureContent = webView.hasOnlySecureContent
+                    self.urlBar.locationView.showTrackingProtectionButton(for: webView.url)
+                }
```

Comment:
@razvanlitianu @nbhasin2 Note: this change is from the original PR referenced, but in practice I'm not seeing this have any noticeable effect, at least not during testing of the main Bugzilla this is meant to address. I will dig in a bit further here, I think that we probably _should_ be hiding the lock icon during load (which is what Safari does AFAICS) but I am not seeing that happening currently so I don't believe this code is having any effect.

Reply:
So far in testing it appears this code for hiding the lock icon is not working entirely, at least for the POC relating to the original Bugzilla. This is the case both for this PR and Razvan's original PR.

For now I'm planning to remove this change and focus on just fixing the lock icon state itself, and then revisit hiding the icon in a follow-up.